n3fit/src/n3fit/hyper_optimization/filetrials.py

```diff
@@ -95,3 +122,21 @@ def new_trial_ids(self, n):
     def new_trial_docs(self, tids, specs, results, miscs):
         self._store_trial = True
         return super().new_trial_docs(tids, specs, results, miscs)
+
+    def to_pkl(self):
+        """Dump `FileTrials` object into a pickle file."""
+        with open(self.pkl_file, "wb") as file:
+            pickle.dump(self, file)
+
+    @classmethod
+    def from_pkl(cls, pickle_filepath):
+        """Load and return an instance of `FileTrials` from a pickle file.
+
+        If a pickle file from previous run is present this method can be used
+            to instantiate an initial `FileTrials` object to restart.
+        """
+        try:
+            with open(pickle_filepath, "rb") as file:
+                return pickle.load(file)
+        except FileNotFoundError as err:
+            log.error("Failed to open pickle file: %s", err)
```

Comment:
I am slightly confused by this. What exactly should be the intended behavior if the file is missing? Should the program halt?

Reply:
If you are doing a restart and `n3fit` cannot not find the file to restart, the program should halt. Thanks for pointing this out! this is not exactly what the `except` part is actually doing in the snippet you highlight. I have corrected the code to now `raise` this exception properly.

Reply:
Great! Thanks.